### ISSUE TITLE
Installation with pip/pip3 fails

### ISSUE THREAD

**Issue by pradeeban** [ZacharyJia/opsdn#1]
$ pip3 --version
pip 22.2 from /Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/pip (python 3.7)
Pradeebans-MacBook-Pro:~ pradeeban$ pip3 install opsdnpy
WARNING: Ignoring invalid distribution -llib3 (/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages)
WARNING: Ignoring invalid distribution -rllib3 (/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages)
WARNING: Ignoring invalid distribution -llib3 (/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages)
WARNING: Ignoring invalid distribution - (/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages)
WARNING: Ignoring invalid distribution -llib3 (/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages)
WARNING: Ignoring invalid distribution -rllib3 (/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages)
WARNING: Ignoring invalid distribution -llib3 (/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages)
WARNING: Ignoring invalid distribution - (/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages)
Collecting opsdnpy
  Using cached opsdnpy-0.2.1-py3-none-any.whl (4.8 kB)
  Using cached opsdnpy-0.2-py3-none-any.whl (4.6 kB)
  Using cached opsdnpy-0.1.3-py3-none-any.whl (4.5 kB)
ERROR: Cannot install opsdnpy==0.1.3, opsdnpy==0.2 and opsdnpy==0.2.1 because these package versions have conflicting dependencies.

The conflict is caused by:
    opsdnpy 0.2.1 depends on networkx==2.8.2
    opsdnpy 0.2 depends on networkx==2.8.2
    opsdnpy 0.1.3 depends on networkx==2.8.2

To fix this you could try to:
1. loosen the range of package versions you've specified
2. remove package versions to allow pip attempt to solve the dependency conflict

ERROR: ResolutionImpossible: for help visit https://pip.pypa.io/en/latest/topics/dependency-resolution/#dealing-with-dependency-conflicts
WARNING: Ignoring invalid distribution -llib3 (/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages)
WARNING: Ignoring invalid distribution -rllib3 (/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages)
WARNING: Ignoring invalid distribution -llib3 (/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages)
WARNING: Ignoring invalid distribution - (/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages)
WARNING: Ignoring invalid distribution -llib3 (/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages)
WARNING: Ignoring invalid distribution -rllib3 (/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages)
WARNING: Ignoring invalid distribution -llib3 (/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages)
WARNING: Ignoring invalid distribution - (/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages)
WARNING: Ignoring invalid distribution -llib3 (/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages)
WARNING: Ignoring invalid distribution -rllib3 (/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages)
WARNING: Ignoring invalid distribution -llib3 (/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages)
WARNING: Ignoring invalid distribution - (/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages)

[notice] A new release of pip available: 22.2 -> 23.0
[notice] To update, run: pip install --upgrade pip

-----------------or----------------------------

$ pip --version
pip 22.2 from /Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/pip (python 3.7)
Pradeebans-MacBook-Pro:~ pradeeban$ pip install opsdnpy
WARNING: Ignoring invalid distribution -llib3 (/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages)
WARNING: Ignoring invalid distribution -rllib3 (/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages)
WARNING: Ignoring invalid distribution -llib3 (/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages)
WARNING: Ignoring invalid distribution - (/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages)
WARNING: Ignoring invalid distribution -llib3 (/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages)
WARNING: Ignoring invalid distribution -rllib3 (/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages)
WARNING: Ignoring invalid distribution -llib3 (/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages)
WARNING: Ignoring invalid distribution - (/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages)
Collecting opsdnpy
  Using cached opsdnpy-0.2.1-py3-none-any.whl (4.8 kB)
Requirement already satisfied: ryu==4.34 in /Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages (from opsdnpy) (4.34)
  Using cached opsdnpy-0.2-py3-none-any.whl (4.6 kB)
  Using cached opsdnpy-0.1.3-py3-none-any.whl (4.5 kB)
ERROR: Cannot install opsdnpy==0.1.3, opsdnpy==0.2 and opsdnpy==0.2.1 because these package versions have conflicting dependencies.

The conflict is caused by:
    opsdnpy 0.2.1 depends on networkx==2.8.2
    opsdnpy 0.2 depends on networkx==2.8.2
    opsdnpy 0.1.3 depends on networkx==2.8.2

To fix this you could try to:
1. loosen the range of package versions you've specified
2. remove package versions to allow pip attempt to solve the dependency conflict

ERROR: ResolutionImpossible: for help visit https://pip.pypa.io/en/latest/topics/dependency-resolution/#dealing-with-dependency-conflicts
WARNING: Ignoring invalid distribution -llib3 (/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages)
WARNING: Ignoring invalid distribution -rllib3 (/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages)
WARNING: Ignoring invalid distribution -llib3 (/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages)
WARNING: Ignoring invalid distribution - (/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages)
WARNING: Ignoring invalid distribution -llib3 (/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages)
WARNING: Ignoring invalid distribution -rllib3 (/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages)
WARNING: Ignoring invalid distribution -llib3 (/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages)
WARNING: Ignoring invalid distribution - (/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages)
WARNING: Ignoring invalid distribution -llib3 (/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages)
WARNING: Ignoring invalid distribution -rllib3 (/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages)
WARNING: Ignoring invalid distribution -llib3 (/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages)
WARNING: Ignoring invalid distribution - (/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages)

[notice] A new release of pip available: 22.2 -> 23.0
[notice] To update, run: pip install --upgrade pip

**Comment by ZacharyJia** [comment]
Hi @pradeeban ,
1. I think the first problem is that you are using python 3.7, while the prebuilt binary is for Python3.8. If you would like to use the prebuilt library, I would suggest you to install a python3.8. Or you have to build the dll-bridge yourself.

2. Could you please check if you have already installed a different version of networkx before? It might be the reason of the dependency problem from pip.

And I just noticed that you are using a Mac? This repo does not provide Mac support because OPNET will not run on Mac. And I also don't have access to a linux version of OPNET, so linux version is also not involved.
In a nutshell, **only windows version** is supported now.

**Comment by pradeeban** [comment]
Thanks, @ZacharyJia for the clarification.

I will close this issue since it is mentioned in the README clearly about the OS and Python requirements.

I do not have access to Windows. But I will proceed with the review.

**Comment by pradeeban** [comment]
Tested `pip install opsdnpy` on Python 3.8 on Ubuntu. It worked.

**Comment by danielskatz** [comment]
note: this was part of https://github.com/openjournals/joss-reviews/issues/4815
